scripts/integration/integration.py

```diff
@@ -486,14 +497,39 @@ def main():
         default='scripts/flamegraph',
         type=Path,
         help='Path to flamegraph script')
+    parser.add_argument(
+        '-v',
+        '--verbosity',
+        default='DEBUG',
+        help='Set the logging verbosity')
 
     args = parser.parse_args()
+
+    # Setup logging.
+    LOGGER.setLevel(logging.DEBUG)
+    fmt = "%(asctime)s %(levelname)-8s [%(name)s] %(message)s"
+    colored_formatter = coloredlogs.ColoredFormatter(fmt)
+    plain_formatter = logging.Formatter(fmt)
+    formatter = colored_formatter if sys.stdout.isatty() else plain_formatter
+    ch = logging.StreamHandler()
+    ch.setLevel(args.verbosity)
+    ch.setFormatter(formatter)
+    LOGGER.addHandler(ch)
+    class ShutdownHandler(logging.Handler):
+        def emit(self, record):
+            logging.shutdown()
+            sys.exit(1)
```

Comment:
Why do we need `sys.exit()` in this handler? Isn't there a possible race with `handle_exit_signal()`?
Please add a comment in case this is intentional.

Reply:
The idea is that a the `critical` log-level automatically terminates the process, e.g., when something fails badly internally. I looks like we could add this instead of `sys.exit(1)`:
```
    signal_subprocs(signal.SIGINT)
    sys.exit(1)
```

...or slightly tweak the shutdown process with a unified function, factored from `handle_exit_signal`.

Reply:
I think in this case we can modify `handle_exit_signal` to:
```python
  signal_subprocs(signum)
  LOGGER.critical('Got signal {signum}, shutting down')
```